src/utils/useSpringAnimation.ts

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import { useSpring } from "@react-spring/web";
+
+export function useSpringAnimation(ref: React.RefObject<HTMLDivElement>) {
+  const [springValues, springRef] = useSpring(
+    () => ({
+      x: 0,
+      y: 0,
+      config: {
+        mass: 2,
+        friction: 14,
+        tension: 120,
+      },
+    }),
+    []
+  );
+  const handleMouseLeave = () => {
+    if (ref.current) {
+      const regex =
+        /translate\((?<offsetX>[-\d.]+)px,\s*(?<offsetY>[-\d.]+)px\)/;
+      const transformValue = ref.current.style.transform;
+      const matches = regex.exec(transformValue);
+
+      if (matches?.groups) {
+        const { offsetX, offsetY } = matches.groups;
+        springRef.start({
+          from: {
+            x: parseFloat(offsetX),
+            y: parseFloat(offsetY),
```

Comment:
This might be paranoid, but still.

As a safety check, I'd put `parseFloat(offsetX) || 0` in both cases, since `parseFloat` may return NaN.
In case some incorrect values for transform make it through or you break the regex accidentally, the animation would still work as expected.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseFloat#return_value
> A floating point number parsed from the given string, or `NaN` when the first non-whitespace character cannot be converted to a number.